rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -99,38 +169,160 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
 
-  // Calculate the wind force.
-  double wind_strength = wind_force_mean_ + wind_force_n_(random_generator_);
-  math::Vector3 wind = wind_strength * wind_direction_;
-  // Apply a force from the constant wind to the link.
-  link_->AddForceAtRelativePosition(wind, xyz_offset_);
-
-  math::Vector3 wind_gust(0, 0, 0);
-  // Calculate the wind gust force.
-  if (now >= wind_gust_start_ && now < wind_gust_end_) {
-    double wind_gust_strength = wind_gust_force_mean_ + wind_gust_force_n_(random_generator_);
-    wind_gust = wind_gust_strength * wind_gust_direction_;
-    // Apply a force from the wind gust to the link.
-    link_->AddForceAtRelativePosition(wind_gust, xyz_offset_);
+  math::Vector3 wind_velocity = math::Vector3(0,0,0);
+
+  if (!custom_static_wind_field_) // Choose method
+  {
+    // Calculate the wind force.
+    double wind_strength = wind_force_mean_ + wind_force_n_(random_generator_);
+    math::Vector3 wind = wind_strength * wind_direction_;
+    // Apply a force from the constant wind to the link.
+    link_->AddForceAtRelativePosition(wind, xyz_offset_);
+
+    math::Vector3 wind_gust(0, 0, 0);
+    // Calculate the wind gust force.
+    if (now >= wind_gust_start_ && now < wind_gust_end_) {
+      double wind_gust_strength = wind_gust_force_mean_ + wind_gust_force_n_(random_generator_);
+      wind_gust = wind_gust_strength * wind_gust_direction_;
+      // Apply a force from the wind gust to the link.
+      link_->AddForceAtRelativePosition(wind_gust, xyz_offset_);
+    }
+
+    geometry_msgs::WrenchStamped wrench_msg;
+
+    wrench_msg.header.frame_id = frame_id_;
+    wrench_msg.header.stamp.sec = now.sec;
+    wrench_msg.header.stamp.nsec = now.nsec;
+    wrench_msg.wrench.force.x = wind.x + wind_gust.x;
+    wrench_msg.wrench.force.y = wind.y + wind_gust.y;
+    wrench_msg.wrench.force.z = wind.z + wind_gust.z;
+    wrench_msg.wrench.torque.x = 0;
+    wrench_msg.wrench.torque.y = 0;
+    wrench_msg.wrench.torque.z = 0;
+
+    wind_pub_.publish(wrench_msg);
+
+    // Calculate the wind speed
+    double wind_speed = wind_speed_mean_; // + wind_speed_n_(random_generator_);
+    wind_velocity = wind_speed * wind_direction_;
   }
+  else
+  {
+    // Get the current position of the aircraft in world coordinates
+    math::Vector3 link_position = link_->GetWorldPose().pos;
+
+    //Determine the wind velocity at the aircraft position if within world bounds. Else, set to user defined constant value.
+    if (!(link_position.x < coords_x_[0] || link_position.x > coords_x_[coords_x_.size()-1] || link_position.y < coords_y_[0] || link_position.y > coords_y_[coords_y_.size()-1] || link_position.z < z_min_ || link_position.z > z_max_))
+    {
+      // Create a vector containing the coordinates of 8 points corresponding to the vertices of the volume enclosing the link and the 5 points for interpolation
+      // Containing as well the wind velocity value for each point.
+      std::vector<math::Vector3> wind_field_cells[2];
```

Comment:
With the change of format for the text file, I am not using this two-element vector anymore.